cmd/airflow.go

```diff
@@ -509,6 +517,24 @@ func airflowInit(cmd *cobra.Command, args []string) error {
 		projectName = strings.Replace(strcase.ToSnake(projectDirectory), "_", "-", -1)
 	}
 
+	if fromTemplate == "select-template" {
+		selectedTemplate, err := selectedTemplate()
+		if err != nil {
+			return fmt.Errorf("unable to select template from list: %w", err)
+		}
+		fromTemplate = selectedTemplate
+	}
+
+	if fromTemplate != "" {
+		templateList, err := TemplateList()
+		if err != nil {
+			return fmt.Errorf("unable to fetch template list: %w", err)
+		}
+		if !isValidTemplate(templateList, fromTemplate) {
+			return fmt.Errorf("%s is not a valid template name. Available templates are: %s", fromTemplate, templateList)
+		}
+	}
```

Comment:
```suggestion
	if fromTemplate == "select-template" {
		selectedTemplate, err := selectedTemplate()
		if err != nil {
			return fmt.Errorf("unable to select template from list: %w", err)
		}
		fromTemplate = selectedTemplate
	} else if fromTemplate != "" {
		templateList, err := TemplateList()
		if err != nil {
			return fmt.Errorf("unable to fetch template list: %w", err)
		}
		if !isValidTemplate(templateList, fromTemplate) {
			return fmt.Errorf("%s is not a valid template name. Available templates are: %s", fromTemplate, templateList)
		}
	}
```
minor nit: we don't need to re-fetch the list and validate the selection from the user

Reply:
Yes, that make sense. Updated the code block.